api/routes.go

```diff
@@ -76,3 +100,28 @@ func tokenFromRequest(r *http.Request) (string, error) {
 	}
 	return value, nil
 }
+
+// userFromRequest returns a user object based on the JWT within the request.
+// Note that this method does not rely on a token being stored in the context.
+func (api *API) userFromRequest(r *http.Request) *database.User {
+	t, err := tokenFromRequest(r)
+	if err != nil {
+		return nil
+	}
+	token, err := jwt.ValidateToken(api.staticLogger, t)
+	if err != nil {
+		return nil
+	}
+	if reflect.ValueOf(token.Claims).Kind() != reflect.ValueOf(jwt2.MapClaims{}).Kind() {
+		return nil
+	}
+	claims := token.Claims.(jwt2.MapClaims)
+	if reflect.ValueOf(claims["sub"]).Kind() != reflect.String {
+		err = errors.New("the token does not contain the sub we expect")
```

Comment:
conversation resolved but code unchanged - forgot to push? - resolved by mistake?

Reply:
Good catch! This was happening at two places and I fixed one of them and somehow didn't fix the other.
Fixed now.